WcaOnRails/app/webpacker/components/RolesTab/RoleForm.jsx

```diff
@@ -0,0 +1,130 @@
+import React from 'react';
+
+import { Form } from 'semantic-ui-react';
+
+import _ from 'lodash';
+import useLoadedData from '../../lib/hooks/useLoadedData';
+import {
+  roleDataUrl,
+  roleUpdateUrl,
+} from '../../lib/requests/routes.js.erb';
+import useSaveAction from '../../lib/hooks/useSaveAction';
+import Errored from '../Requests/Errored';
+import DelegateForm from './DelegateForm';
+import Loading from '../Requests/Loading';
+
+const groups = [{
+  key: 'delegates',
+  text: 'Delegates',
+  value: 'Delegates',
+}];
+
+const delegateStatusOptions = ['trainee_delegate', 'candidate_delegate', 'delegate', 'senior_delegate'];
+
+export default function RoleForm({ userId, roleId }) {
+  const { data, loading, error } = useLoadedData(roleDataUrl(userId, roleId));
+  const { save } = useSaveAction();
+  const selectedGroup = groups[0].value;
+  const [formValues, setFormValues] = React.useState({});
+  const [apiError, setError] = React.useState(false);
+  const [saving, setSaving] = React.useState(false);
+  const [finished, setFinished] = React.useState(false);
+
+  React.useEffect(() => {
+    setFormValues({
+      delegateStatus: delegateStatusOptions[0],
+      location: '',
+      ...(data?.roleData || {}),
+    });
+  }, [data]);
+
+  const updateRole = () => {
+    setSaving(true);
```

Comment:
The `useSaveAction` hook returns its own `saving` state. You do not have to manage it manually